File: Python/f/f7-Adivinanzas.py
'''
7. Simula un juego de adivinanzas
'''
import random

adiv = random.randint(1,20)
while True:
    num = int(input("Adivina el numero:"))
    print(adiv)
    if num == adiv:
        print("Ganaste")
        break
    elif num < adiv:
        print("Tu numero es menor")
        pista = random.randint(num,adiv)
        while pista == adiv or pista == num:
            pista = random.randint(num,adiv)
        print("Pista: {}", pista)
    else:
        print("Tu numero es mayor")
        pista = random.randint(adiv,num)
        while pista == adiv or pista == num:
            pista = random.randint(adiv,num)
        print("Pista: {}", pista)

File: Rust/f/f7-Adivinanzas.rs
// 7. Simula un juego de adivinanzas
use std::io;
use rand::Rng;
fn main(){
    let adiv:i32 = 7;

    loop {

        let mut n = String::new();
        println!("Introduce un número: ");
        let _ = io::stdin().read_line(&mut n);
        let r:i32 = n.trim().parse().expect("Por favor introduce un número");
        if r == adiv {
            break;
        }else if r > adiv && r != 20 {
            println!("El número es mayor");
            let mut rng = rand::thread_rng();
        let hint_val: i32 = rng.gen_range(1..=20);
            println!("Pista: {}", hint_val);
        }else if r < adiv && r != 1 {
            println!("El número es menor");
        }
    }
}
